packages/material-ui-lab/src/DataGrid/DataGrid.js

```diff
@@ -159,13 +159,18 @@ const defaultDataProviderFactory = ({ rowsData, defaultColumnOptions, columnsKey
         }, null);
       });
     }
-
-    return newRowsData;
-  },
+    resolve(newRowsData.splice(params.pagination.paginationPage * params.pagination.paginationPageSize, params.pagination.paginationPageSize));
```

Comment:
Pagination won't always be used, e.g. infinite scroll. I think that we need a code branch.

Reply:
Do you want to split it by the pagination prop or how would you split it?

Reply:
I was thinking of something like this:
```suggestion
    if (PAGINATION NEEDED) {
       /* cut the data were needed to only provide the piece the UI ask for. */
    }

    resolve(newRowsData);
```

Reply:
Then, the component will have to append the date to its datastore, in the future, we can look into caching and >1 requests in parallels.

Reply:
Basically, we should consider that this getList call is an SQL query that goes to the network even if we work on the client only (for now).

Reply:
I came to the same conclusion after my commit and implemented exactly what you proposed ;) we should try to make it generic as possible. I will think about a good approach.